### PR TITLE
Fix savestates in Bluetooth passthrough mode

### PR DESCRIPTION
**Pull request by leoetlino**
This fixes savestates when using Bluetooth passthrough by keeping track of pending transfer commands and discarding them on state load, so that the emulated software receives a reply to IOS requests as expected.

With this change, savestates in BT passthrough should work as noted in the following table:

| BT state when saving | when loading | Result |
| --- | --- | --- |
| N remotes connected | N remotes connected **during the same session**<br>(remotes were *not* disconnected after saving) |  Working :white_check_mark: |
| N remotes connected | some or all remotes disconnected | Working\* :white_check_mark: |
| 0 remote connected | 0 remote connected | Working :white_check_mark: |
| N remotes connected | some or all remotes connected during a different session<br>(i.e., disconnected after saving) | Not supported |
| 0 remote connected | 1 or more remotes connected | Not supported |

<sup>\* On the first connection attempt, the game will realise that the Wiimote has disconnected. A remote can then be connected. Also, you may need to connect then disconnect a Wiimote first before loading the state; otherwise the adapter may get confused.</sup>

Basically, as long as there's no Bluetooth connection, or the Bluetooth connections all match (which are the most common cases), then it should work fine.

This PR also gets rid of the static state, which was previously needed because the transfer callbacks were static methods.

### CODE REVIEW

**Comment by leoetlino** [comment]
Temporarily closing while I investigate something I broke

**Comment by leoetlino** [comment]
This should now be working as explained in the PR description.

**Comment by mbc07** [comment]
What happens if we try to load a state made with a emulated Wiimote in passthrough mode or vice-versa?

**Comment by leoetlino** [comment]
You can't. Dolphin will detect it and prevent the state from loading.
